go/service/identify.go

```diff
@@ -147,8 +150,229 @@ func (h *IdentifyHandler) resolveUserOrTeam(ctx context.Context, arg string) (u
+	if err != nil {
+		return res, err
+	}
+	return h.resolveIdentifyImplicitTeamHelper(ctx, arg, writerAssertions, readerAssertions)
+}
+
+func (h *IdentifyHandler) resolveIdentifyImplicitTeamHelper(ctx context.Context, arg keybase1.ResolveIdentifyImplicitTeamArg,
+	writerAssertions, readerAssertions []libkb.AssertionExpression) (res keybase1.ResolveIdentifyImplicitTeamRes, err error) {
+
+	lookupName := keybase1.ImplicitTeamDisplayName{
+		IsPublic: arg.IsPublic,
+	}
+	if len(arg.Suffix) > 0 {
+		lookupName.ConflictInfo, err = libkb.ParseImplicitTeamDisplayNameSuffix(arg.Suffix)
+		if err != nil {
+			return res, err
+		}
+	}
+
+	var resolvedAssertions []resolvedAssertion
+
+	err = h.resolveImplicitTeamSet(ctx, writerAssertions, &lookupName.Writers, &resolvedAssertions)
+	if err != nil {
+		return res, err
+	}
+	err = h.resolveImplicitTeamSet(ctx, readerAssertions, &lookupName.Readers, &resolvedAssertions)
+	if err != nil {
+		return res, err
+	}
+
+	lookupNameStr, err := teams.FormatImplicitTeamDisplayName(ctx, h.G(), lookupName)
+	if err != nil {
+		return res, err
+	}
+	h.G().Log.CDebugf(ctx, "ResolveIdentifyImplicitTeam looking up:'%v'", lookupNameStr)
+
+	var teamID keybase1.TeamID
+	var impName keybase1.ImplicitTeamDisplayName
+	// Lookup*ImplicitTeam is responsible for making sure the returned team has the members from lookupName.
+	// Duplicates are also handled by Lookup*. So we might end up doing extra identifies of duplicates out here.
+	// (Duplicates e.g. "me,chris,chris", "me,chris#chris", "me,chris@rooter#chris")
+	if arg.Create {
+		teamID, impName, err = teams.LookupOrCreateImplicitTeam(ctx, h.G(), lookupNameStr, arg.IsPublic)
+	} else {
+		teamID, impName, err = teams.LookupImplicitTeam(ctx, h.G(), lookupNameStr, arg.IsPublic)
+	}
+	if err != nil {
+		return res, err
+	}
+
+	// can be nil
+	myUID := h.G().ActiveDevice.UID()
+
+	var displayNameKBFS string
+	if myUID.Exists() {
+		var name libkb.NormalizedUsername
+		name, err = h.G().GetUPAKLoader().LookupUsername(ctx, myUID)
+		if err != nil {
+			return res, err
+		}
+		// display name with the logged-in user first
+		displayNameKBFS, err = teams.FormatImplicitTeamDisplayNameWithUserFront(ctx, h.G(), impName, name)
+	} else {
+		displayNameKBFS, err = teams.FormatImplicitTeamDisplayName(ctx, h.G(), impName)
+	}
+	if err != nil {
+		return res, err
+	}
+
+	team, err := teams.Load(ctx, h.G(), keybase1.LoadTeamArg{
+		ID:          teamID,
+		ForceRepoll: true,
+	})
+	if err != nil {
+		return res, err
+	}
+
+	writers, err := team.UsersWithRoleOrAbove(keybase1.TeamRole_WRITER)
+	if err != nil {
+		return res, err
+	}
+
+	// Populate the result. It may get returned together with an identify error.
+	res = keybase1.ResolveIdentifyImplicitTeamRes{
+		DisplayName: displayNameKBFS,
+		TeamID:      teamID,
+		Writers:     writers,
+		TrackBreaks: nil,
+	}
+
+	if arg.DoIdentifies {
+		return h.resolveIdentifyImplicitTeamDoIdentifies(ctx, arg, res, resolvedAssertions)
```

Comment:
is there a problem if the identifies fail after a create?

Reply:
That's an interesting point. But I think that the semantics of this are supposed are that the IDs are icing on the cake. In the other system it would have been separate Resolve then Identify. And part of the resolve response is the TeamID, which we can only get by creating the iteam. Soo.. could ask strib, cause it's really about how they want to use it?

Reply:
It is possible to reverse the operations and to the ID before the creation?

Reply:
Yesbut KBFS might be expecting the rest of the response to be valid even if an ID fails. I asked on confluence

Reply:
ok, then leave as is, thanks